### DISCUSSION TITLE
@walletconnect/modal-react-native not working

### DISCUSSION THREAD

**Discussion by navdeepgarg07** [WalletConnect/walletconnect-monorepo#2870] [Developer Support]
I am using packing @walletconnect/modal-react-native in react-native for V2 migration. But i am getting init connection failed error. How to ressolve this issue please help me out?

error: {"context": "core"} Native module not found
 INFO  {"context": "client"} SignClient Initialization Failure
 ERROR  {"context": "client"} Native module not found

**Comment by ignaciosantise** [comment]
hey @navdeepgarg07 👋  did you follow the installation & usage steps from the documentation? You can read it here: https://docs.walletconnect.com/2.0/reactnative/walletconnectmodal/installation

Also, you can check the implementation in our example repo: https://github.com/WalletConnect/react-native-examples/blob/main/dapps/v2Explorer/src/screens/App.tsx

**Reply by navdeepgarg07** [reply]
![Screenshot 2023-06-30 at 1 15 22 PM](https://github.com/WalletConnect/walletconnect-monorepo/assets/136788312/aaacd44f-8aaf-41c9-b61a-574c064b4c32)
I have followed all the steps but still getting this issue. Please check the attached screenshot.

**Reply by ignaciosantise** [reply]
@navdeepgarg07 can you upload an example repository so i can reproduce error?

Does the error also happen to you in this project? https://github.com/WalletConnect/react-native-examples/blob/main/dapps/v2Explorer

**Comment by Lestonz** [comment]
Hello @ignaciosantise , 
I am using this library @walletconnect/modal-react-native for my app. In IOS is working Metamask, Trust, Rainbow etc. (Mainnet and Goerli), but Android is not connecting Metamask, that is open but has no sign-in message or transaction message. Rainbow is working for Android. In addition, I am using web3.js library for my app, `const { provider } = useWalletConnectModal();` is not working good. For example when I asked ETH balance giving error. 

my code; 
```
  const { open, provider, isConnected, address } = useWalletConnectModal();
  const web3Provider = useMemo(
      () => new Web3(provider ? new Web3(provider) : undefined), [provider]
  );
    const getMyBalance = async () => {
        const myBalance = await web3.eth.getBalance(address);
        console.log("My Eth Balance: ", myBalance)
    }
```

Best regards,
Melih
